### ISSUE TITLE
npm audit report: semver vulnerable semver  <7.5.2

### ISSUE THREAD

**Issue by pawan-gwebs** [angular/angular#50900]
### Which @angular/* package(s) are the source of the bug?

compiler-cli

### Is this a regression?

No

### Description

# npm audit report

semver  <7.5.2
Severity: moderate
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
fix available via `npm audit fix --force`
Will install @angular/compiler-cli@10.2.5, which is a breaking change
node_modules/@babel/core/node_modules/semver
node_modules/@babel/helper-compilation-targets/node_modules/semver
node_modules/@babel/helper-create-class-features-plugin/node_modules/semver
node_modules/@babel/helper-create-regexp-features-plugin/node_modules/semver
node_modules/@babel/helper-define-polyfill-provider/node_modules/semver
node_modules/@babel/plugin-transform-runtime/node_modules/semver
node_modules/@babel/preset-env/node_modules/semver
node_modules/babel-plugin-polyfill-corejs2/node_modules/semver
node_modules/istanbul-lib-instrument/node_modules/semver
node_modules/less/node_modules/semver
node_modules/make-dir/node_modules/semver
  @babel/core  *
  Depends on vulnerable versions of @babel/helper-compilation-targets
  Depends on vulnerable versions of semver
  node_modules/@babel/core
    @angular-devkit/build-angular  0.6.6 || >=0.803.0-next.0
    Depends on vulnerable versions of @angular/compiler-cli
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of @babel/plugin-proposal-async-generator-functions
    Depends on vulnerable versions of @babel/plugin-transform-async-to-generator
    Depends on vulnerable versions of @babel/plugin-transform-runtime
    Depends on vulnerable versions of @babel/preset-env
    Depends on vulnerable versions of @ngtools/webpack
    Depends on vulnerable versions of babel-loader
    Depends on vulnerable versions of babel-plugin-istanbul
    Depends on vulnerable versions of less
    node_modules/@angular-devkit/build-angular
    @angular/compiler-cli  >=11.0.0-next.0
    Depends on vulnerable versions of @babel/core
    node_modules/@angular/compiler-cli
      @ngtools/webpack  >=11.0.3
      Depends on vulnerable versions of @angular/compiler-cli
      node_modules/@ngtools/webpack
    @babel/helper-compilation-targets  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/@babel/helper-compilation-targets
      @babel/plugin-transform-classes  >=7.19.0
      Depends on vulnerable versions of @babel/helper-compilation-targets
      node_modules/@babel/plugin-transform-classes
        @babel/preset-env  *
        Depends on vulnerable versions of @babel/helper-compilation-targets
        Depends on vulnerable versions of @babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression
        Depends on vulnerable versions of @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining
        Depends on vulnerable versions of @babel/plugin-syntax-unicode-sets-regex
        Depends on vulnerable versions of @babel/plugin-transform-async-generator-functions
        Depends on vulnerable versions of @babel/plugin-transform-async-to-generator
        Depends on vulnerable versions of @babel/plugin-transform-class-properties
        Depends on vulnerable versions of @babel/plugin-transform-class-static-block
        Depends on vulnerable versions of @babel/plugin-transform-classes
        Depends on vulnerable versions of @babel/plugin-transform-dotall-regex
        Depends on vulnerable versions of @babel/plugin-transform-function-name
        Depends on vulnerable versions of @babel/plugin-transform-named-capturing-groups-regex
        Depends on vulnerable versions of @babel/plugin-transform-object-rest-spread
        Depends on vulnerable versions of @babel/plugin-transform-private-methods
        Depends on vulnerable versions of @babel/plugin-transform-private-property-in-object
        Depends on vulnerable versions of @babel/plugin-transform-unicode-property-regex
        Depends on vulnerable versions of @babel/plugin-transform-unicode-regex
        Depends on vulnerable versions of @babel/plugin-transform-unicode-sets-regex
        Depends on vulnerable versions of babel-plugin-polyfill-corejs2
        Depends on vulnerable versions of babel-plugin-polyfill-corejs3
        Depends on vulnerable versions of babel-plugin-polyfill-regenerator
        Depends on vulnerable versions of semver
        node_modules/@babel/preset-env
      @babel/plugin-transform-function-name  >=7.16.7
      Depends on vulnerable versions of @babel/helper-compilation-targets
      node_modules/@babel/plugin-transform-function-name
      @babel/plugin-transform-object-rest-spread  *
      Depends on vulnerable versions of @babel/helper-compilation-targets
      node_modules/@babel/plugin-transform-object-rest-spread
    @babel/helper-create-class-features-plugin  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/@babel/helper-create-class-features-plugin
      @babel/plugin-transform-class-properties  *
      Depends on vulnerable versions of @babel/helper-create-class-features-plugin
      node_modules/@babel/plugin-transform-class-properties
      @babel/plugin-transform-private-methods  *
      Depends on vulnerable versions of @babel/helper-create-class-features-plugin
      node_modules/@babel/plugin-transform-private-methods
      @babel/plugin-transform-private-property-in-object  *
      Depends on vulnerable versions of @babel/helper-create-class-features-plugin
      node_modules/@babel/plugin-transform-private-property-in-object
    @babel/helper-create-regexp-features-plugin  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/@babel/helper-create-regexp-features-plugin
      @babel/plugin-transform-dotall-regex  >=7.21.4-esm
      Depends on vulnerable versions of @babel/helper-create-regexp-features-plugin
      node_modules/@babel/plugin-transform-dotall-regex
      @babel/plugin-transform-named-capturing-groups-regex  *
      Depends on vulnerable versions of @babel/core
      Depends on vulnerable versions of @babel/helper-create-regexp-features-plugin
      node_modules/@babel/plugin-transform-named-capturing-groups-regex
      @babel/plugin-transform-unicode-property-regex  *
      Depends on vulnerable versions of @babel/helper-create-regexp-features-plugin
      node_modules/@babel/plugin-transform-unicode-property-regex
      @babel/plugin-transform-unicode-regex  >=7.21.4-esm
      Depends on vulnerable versions of @babel/helper-create-regexp-features-plugin
      node_modules/@babel/plugin-transform-unicode-regex
    @babel/helper-remap-async-to-generator  >=7.18.6
    Depends on vulnerable versions of @babel/core
    node_modules/@babel/helper-remap-async-to-generator
      @babel/plugin-proposal-async-generator-functions  >=7.18.6
      Depends on vulnerable versions of @babel/helper-remap-async-to-generator
      node_modules/@babel/plugin-proposal-async-generator-functions
      @babel/plugin-transform-async-generator-functions  *
      Depends on vulnerable versions of @babel/helper-remap-async-to-generator
      node_modules/@babel/plugin-transform-async-generator-functions
      @babel/plugin-transform-async-to-generator  >=7.18.6
      Depends on vulnerable versions of @babel/helper-remap-async-to-generator
      node_modules/@babel/plugin-transform-async-to-generator
    @babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression  *
    Depends on vulnerable versions of @babel/core
    node_modules/@babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression
    @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining  *
    Depends on vulnerable versions of @babel/core
    node_modules/@babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining
    @babel/plugin-syntax-unicode-sets-regex  *
    Depends on vulnerable versions of @babel/core
    node_modules/@babel/plugin-syntax-unicode-sets-regex
    @babel/plugin-transform-class-static-block  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of @babel/helper-create-class-features-plugin
    node_modules/@babel/plugin-transform-class-static-block
    @babel/plugin-transform-unicode-sets-regex  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of @babel/helper-create-regexp-features-plugin
    node_modules/@babel/plugin-transform-unicode-sets-regex
    babel-loader  >=8.0.0-beta.0
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of find-cache-dir
    node_modules/babel-loader
    istanbul-lib-instrument  >=1.2.0
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/istanbul-lib-instrument
      babel-plugin-istanbul  >=3.1.0-candidate.0
      Depends on vulnerable versions of istanbul-lib-instrument
      node_modules/babel-plugin-istanbul
      karma-coverage  >=2.0.0
      Depends on vulnerable versions of istanbul-lib-instrument
      Depends on vulnerable versions of istanbul-lib-report
      Depends on vulnerable versions of istanbul-reports
      node_modules/karma-coverage
  @babel/helper-define-polyfill-provider  *
  Depends on vulnerable versions of @babel/core
  Depends on vulnerable versions of @babel/helper-compilation-targets
  Depends on vulnerable versions of semver
  node_modules/@babel/helper-define-polyfill-provider
    babel-plugin-polyfill-corejs3  >=0.0.1
    Depends on vulnerable versions of @babel/helper-define-polyfill-provider
    node_modules/babel-plugin-polyfill-corejs3
    babel-plugin-polyfill-regenerator  >=0.0.1
    Depends on vulnerable versions of @babel/helper-define-polyfill-provider
    node_modules/babel-plugin-polyfill-regenerator
  @babel/plugin-transform-runtime  >=7.1.0
  Depends on vulnerable versions of babel-plugin-polyfill-corejs2
  Depends on vulnerable versions of babel-plugin-polyfill-corejs3
  Depends on vulnerable versions of babel-plugin-polyfill-regenerator
  Depends on vulnerable versions of semver
  node_modules/@babel/plugin-transform-runtime
  babel-plugin-polyfill-corejs2  >=0.0.1
  Depends on vulnerable versions of @babel/helper-define-polyfill-provider
  Depends on vulnerable versions of semver
  node_modules/babel-plugin-polyfill-corejs2
  make-dir  2.0.0 - 3.1.0
  Depends on vulnerable versions of semver
  node_modules/less/node_modules/make-dir
  node_modules/make-dir
    find-cache-dir  2.1.0 - 3.3.2
    Depends on vulnerable versions of make-dir
    node_modules/find-cache-dir
    istanbul-lib-report  >=2.0.5
    Depends on vulnerable versions of make-dir
    node_modules/istanbul-lib-report
      istanbul-reports  >=3.0.0-alpha.0
      Depends on vulnerable versions of istanbul-lib-report
      node_modules/istanbul-reports
    less  >=3.11.2
    Depends on vulnerable versions of make-dir
    node_modules/less

42 moderate severity vulnerabilities

### Please provide the exception or error you saw

```true
semver  <7.5.2
Severity: moderate
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
fix available via `npm audit fix --force`
Will install @angular/compiler-cli@10.2.5, which is a breaking change
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
^16.1.0
```

### Anything else?

We are getting 42 moderate severity vulnerabilities on new angular project creation and almost 45 vulnerabilities on existing project after updated to version 16.1.3.
All are related to "semver  <7.5.2" mentioned in different dependencies in package-lock.json file.

**Comment by rcb98** [comment]
This also affects the `localize` package:
```
semver  <7.5.2
Severity: moderate
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
No fix available
node_modules/@angular/compiler-cli/node_modules/@babel/core/node_modules/semver
node_modules/@angular/localize/node_modules/semver
```
Is is possible to get it upgraded for the v.15 as well? Since it is still supported.

**Comment by JoostK** [comment]
It cannot be fully avoided at the moment, because of babel/babel#15720. The CLI has already updated its usage of `semver` to 7.5.3, but the transitive usage through Babel cannot be avoided.

If you are concerned by this CVE and would like to avoid, then you'll have to use `overrides`/`resolutions` in your package.json to force-resolve a non-vulnerable version of `semver`.

**Comment by gon250** [comment]
Ive the same issue here 👆🏼

**Comment by JeanMeche** [comment]
Babel has merged a fix babel/babel#15742, we're waiting for their release to update our deps !

**Comment by JoostK** [comment]
Closing as Babel has since been updated.
